File: 1752-check-if-array-is-sorted-and-rotated/1752. Check if Array Is Sorted and Rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        lst = deque(nums)
        n = len(nums)
        for i in range(n):
            if lst == deque(sorted(lst)):
                return True
            lst.append(lst.popleft())
            print(lst, sorted(lst), lst == sorted(lst))
        return False

File: 1752-check-if-array-is-sorted-and-rotated/1752. Check if Array Is Sorted and Rotated.java
class Solution {
    public boolean check(int[] nums) {
        int decreases = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > nums[((i+1)+nums.length) % nums.length]) {
                decreases++;
            }
        }
        return decreases<=1;
    }
}